### PR TITLE
[FIX] hr_work_entry_holidays: ensure the consistency of leaves

### PR DESCRIPTION
**Pull request by thle-odoo**
Steps to reproduce:
-------------------
- install `hr_work_entry_holidays` module
- create an employee and add two contracts:
    - from 2023-01-01 to 2023-06-30 with a full time (5/5) resource which is expired
    - from 2023-07-01 to 2023-12-31 with a partial time (4/5) resource which is running (doesn't work on Wednesday)
- create an allocation with 10 days
- with the employee, create two leaves:
    - 3 days during the first semester with one Wednesday
    - 3 days during the second semester with one Wednesday

Issue:
------
Leave duration is based on the current contract.
If we change the type of leave by modifying its days/hours unit, we will get inconsistencies between days and hours for leave taken in a period belonging to another contract.

Cause:
------
Expired contracts are not taken into account
when calculating the number of days and hours.

Solution:
---------
Add `close` state to contract search.

opw-3419380

### CODE REVIEW

File: addons/hr_work_entry_holidays/tests/test_multi_contract.py [hunk 1]
```diff
@@ -169,3 +169,105 @@ def test_leave_multi_contracts_split(self):
...
+                'date_from': datetime.combine(date(2023, 1, 3), time.min), # Tuesday
+                'date_to': datetime.combine(date(2023, 1, 5), time.max), # Thursday
+            },
+            {
+                'employee_id': employee.id,
+                'holiday_status_id': leave_type.id,
+                'number_of_days': 2,
+                'date_from': datetime.combine(date(2023, 12, 5), time.min), # Tuesday
+                'date_to': datetime.combine(date(2023, 12, 7), time.max), # Thursday
+            },
```
Comment (Bertrand2) [inline addons/hr_work_entry_holidays/tests/test_multi_contract.py:256]: Should we enforce the number of days if the goal is to have them computed correctly?

**Review by Bertrand2 — COMMENTED** [review]
Just a small question for the test, otherwise LGTM :)

File: addons/hr_work_entry_holidays/tests/test_multi_contract.py (hunk 1 above)
Comment (robcoekaerts) [reply addons/hr_work_entry_holidays/tests/test_multi_contract.py:256]: @Bertrand2 When programmatically creating time offs like this it is currently indeed still necessary to 'hardcode' the number of days. This is one of the things my ongoing refactor will fix :)

**Comment by sofiagvaladze** [comment]
robodoo r+

**Pull request closed without merging**